tracing/AsyncWrap/conceptual_overview.md

```diff
@@ -0,0 +1,73 @@
+## Concepts
+
+async\_hooks provides a mechanism to store context at the time of registration of a task (callback) and to restore context prior to execution of that task. This makes it possible to provide a consistent logical context across otherwise disjointed tasks.
+
+When a task (callback) is registered, a context ID is assigned and provided as a parameter to "init" hooks. Before the task (callback) is invoked the same context ID is provided to "before" hooks. "init" hooks execute immediately in the registering context and therefore can save off aspects of it; "before" hooks execute in the task invocation context and therefore can restore saved aspects of the registering context prior to that task's execution.
+
+"after" hooks are called immediately after execution of the task and receive the context ID to allow context cleanup. "destroy" hooks are called on destruction of the context to allow resource cleanup.
+
+As implied by the previous paragraph, async\_hooks itself only manages a context ID, not actual context. It is up to a hook author to associate data with this ID in an init hook, for example in a persistent map of context IDs to context objects.
+
+## Terms
+
+**async\_id** is a unique ID assigned to every context. An incrementing integer; 1 represents the root context.
```

Comment:
Depending on the definition of context used, a single context may capture the work associated with multiple async ids. I think we will have to be careful with the word "context" here (or give a precise definition for its use in the doc).